src/components/Dialog-Window/Resources/DialogResources.tsx

```diff
@@ -1,44 +1,83 @@
+import { v4 as uuidV4 } from "uuid";
+import { Cross2Icon } from "@radix-ui/react-icons";
 import * as React from "react";
 import styles from "./DialogResources.module.scss";
+import { useLocalStorage } from "../../../h5p/H5P.util";
+import { Link } from "../../../types/Link";
 
 export type DialogResourceProps = {
   relevantLinks: string[];
-  customLinks: string[];
+  id: string;
 };
 
 export const DialogResources: React.FC<DialogResourceProps> = ({
   relevantLinks,
-  customLinks,
+  id,
 }) => {
-  const customItems = customLinks.map((link: string) => (
-    <li key={link} className={styles.li}>
-      <a href={link}>{link}</a>
-    </li>
-  ));
+  const [currentLocalStorage, setCurrentLocalStorage] = useLocalStorage(
+    "h5p-topic-map-userdata",
+    id,
+  );
+  const [link, setLink] = React.useState("");
+  const [customLinks, setCustomLinks] = React.useState<Array<JSX.Element>>([]);
+  const inputFieldRef = React.useRef<HTMLInputElement>(null);
 
-  const relevantItems = relevantLinks.map((link: string) => (
-    <li key={link} className={styles.li}>
-      <a href={link}>{link}</a>
+  const removeCustomLink = (linkToRemove: string): void => {
+    currentLocalStorage[id].links = currentLocalStorage[id].links?.filter(
+      (item: Link) => item.id !== linkToRemove,
+    );
+
+    setCurrentLocalStorage(currentLocalStorage);
+    // we can disable this check since this function will not be called before the page is rendered
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    populateCustomLinks();
+  };
+
+  const relevantItems = relevantLinks.map((item: string) => (
+    <li key={item} className={styles.li}>
+      <a href={item}>{item}</a>
     </li>
   ));
 
-  const [cList, setList] = React.useState(customItems);
-  const [link, setLink] = React.useState("");
-  const inputFieldRef = React.useRef<HTMLInputElement>(null);
+  // extract the generation of custom links list to separate function
+  const populateCustomLinks = (): void => {
+    if (currentLocalStorage[id].links) {
+      setCustomLinks((): JSX.Element[] => {
+        return currentLocalStorage[id].links!.map((item: Link) => (
```

Comment:
Is this forced non-null assertion necessary, as we null check it?

```suggestion
        return currentLocalStorage[id].links.map((item: Link) => (
```

Reply:
Yes! ✅